lib/dynamodb/converter.js

```diff
@@ -221,7 +223,16 @@ function formatList(data, options) {
  * @param wrapNumbers [Boolean]
  */
 function convertNumber(value, wrapNumbers) {
-  return wrapNumbers ? new NumberValue(value) : Number(value);
+  if (wrapNumbers) {
+    return new NumberValue(value);
+  }
+  try {
+    return Number.isSafeInteger(value) ? Number(value) : BigInt(value);
+  } catch (err) {
+    if (err === 'BigInt is not defined') {
```

Comment:
@AllanFly120 Do you have any way that the SDK could throw a warning to the user that the number they're deserializing is not a safe int, and to use the wrap option?

Reply:
Ignore the BigInt error check. This is not valid. 

But still curious, does the AWS SDK throw any warning messages? I noticed before if you did not provide the wrapNumbers option that you could actually get an unsafe number back? 
his is interesting. Say, I used the UI and added a Number to the dynamo table. If I queried the table and did not pass that option it looks like it'd return an unsafe casted number of that value and not warn the user of that. Personally,as a consumer of this SDK I'd appreciate some kind of warning and prompting to use that option if this happened.

Reply:
@AllanFly120

Reply:
@TechTeam12 Can you create a separate issue for discussing this?

Reply:
Will do!

Reply:
Little long winded, but: 
https://github.com/aws/aws-sdk-js/issues/3065